kuryr_kubernetes/controller/drivers/lbaasv2.py

```diff
@@ -1084,8 +1119,16 @@ def _wait_for_deletion(self, loadbalancer, timeout,
 
         for remaining in self._provisioning_timer(timeout, interval):
             try:
-                lbaas.get_load_balancer(loadbalancer.id)
-            except o_exc.ResourceNotFound:
+                lb = lbaas.get_load_balancer(loadbalancer.id)
+                if lb.provisioning_status not in ('DELETED', 'PENDING_DELETE'):
+                    # Octavia tends to ignore delete requests, so just screw it
+                    # and retry if that happens.
+                    LOG.warning('LB %s does not seem to be deleted. This is '
+                                'unexpected and may indicate a problem with '
+                                'Octavia, retrying!',
+                                loadbalancer.id)
+                    lbaas.delete_load_balancer(loadbalancer.id, cascade=True)
```

Comment:
does it eventually gets deleted?

Reply:
Most of the time yes. I did stress testing on LB with this patch.